templates/kubernetes/terraform/modules/kubernetes/fileupload/main.tf

```diff
@@ -0,0 +1,27 @@
+# File uploads
+
+data "aws_secretsmanager_secret" "cf_keypair" {
+  name = "${var.project}_cf_keypair"
```

Comment:
You need to define variables for this module.

Reply:
I was hoping it would "pass through". :blush:

Reply:
No, all modules are equal. Whether it's defined in a subdirectory, somewhere else on disk, or an external repo, they all have the same definition and requirements. You can think of them as functions, where input variables are the function signature and output variables are the return values. The only thing that is ever passed implicitly is providers.

Reply:
Ah, that's what I was thinking of. Thanks for explaining it. :+1: